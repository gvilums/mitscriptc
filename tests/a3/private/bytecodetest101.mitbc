function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = ["Otherprint: ", None],
			parameter_count = 1,
			local_vars = [s],
			local_ref_vars = [],
			free_vars = [],
			names = [_print],
			instructions = 
			[
				load_global	0
				load_const	0
				load_local	0
				add
				call	1
				pop
				load_const	1
				return
			]
		},
		function
		{
			functions =
			[
				function
				{
					functions = [],
					constants = [": ", None],
					parameter_count = 1,
					local_vars = [s],
					local_ref_vars = [],
					free_vars = [oldprint, d],
					names = [intcast],
					instructions = 
					[
						push_ref	0
						load_ref
						load_global	0
						push_ref	1
						load_ref
						call	1
						load_const	0
						add
						load_local	0
						add
						call	1
						pop
						load_const	1
						return
					]
				}
			],
			constants = [None, "", 1],
			parameter_count = 0,
			local_vars = [d, oldprint],
			local_ref_vars = [oldprint, d],
			free_vars = [],
			names = [print, depth],
			instructions = 
			[
				load_const	0
				store_local	0
				load_const	0
				store_local	1
				load_global	1
				load_const	1
				add
				store_local	0
				load_global	0
				store_local	1
				load_func	0
				push_ref	0
				push_ref	1
				alloc_closure	2
				store_global	0
				load_global	1
				load_const	2
				add
				store_global	1
				load_const	0
				return
			]
		}
	],
	constants = [0, "Should print: Otherprint: hello", "hello", "Should print: Otherprint: 0: 1: 2: 3: hello"],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [_print, print, depth, add_depth, input, intcast],
	instructions = 
	[
		load_func	0
		alloc_closure	0
		store_global	1
		load_func	1
		alloc_closure	0
		store_global	4
		load_func	2
		alloc_closure	0
		store_global	5
		load_global	1
		store_global	0
		load_func	3
		alloc_closure	0
		store_global	1
		load_const	0
		store_global	2
		load_func	4
		alloc_closure	0
		store_global	3
		load_global	0
		load_const	1
		call	1
		pop
		load_global	1
		load_const	2
		call	1
		pop
		load_global	3
		call	0
		pop
		load_global	3
		call	0
		pop
		load_global	3
		call	0
		pop
		load_global	3
		call	0
		pop
		load_global	0
		load_const	3
		call	1
		pop
		load_global	1
		load_const	2
		call	1
		pop
	]
}