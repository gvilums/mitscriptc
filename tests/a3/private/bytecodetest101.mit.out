Should print: Otherprint: hello
Otherprint: hello
Should print: Otherprint: 0: 1: 2: 3: hello
Otherprint: 0: 1: 2: 3: hello
